### ISSUE TITLE
[FEATURE_REQUEST] Summarize: Insert afer last summarized message

### ISSUE THREAD

**Issue by Quairon-Nailo** [SillyTavern/SillyTavern#2638]
### Have you searched for similar requests?

Yes

### Describe the solution you'd like

Title, it'd be nice if the summary could be inserted directly after the events it summarizes, that way we can put it as a "summary of the events before this point", and whatever comes after that summary are things that happened chronologically after the events in the summary, leading to less confusion from the LLM.

### Priority

Medium (Would be very useful)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

**Comment by Wolfsblvt** [comment]
If you want to have more control over the summary and where it's placed, maybe it makes sense for you to consider using STscript to create summaries yourself?

There are some big scripts already available and shared on the Discord. Just look for "Super Summary".

**Comment by Quairon-Nailo** [comment]
That script doesn't really give control over where the summary is placed in the prompt, nor do I see a good way in STscript to do so, since I can't use a variable in @ depth placement, and even if I could do that, I'd need to recalculate the variable after every message (and every message deletion).

**Comment by Wolfsblvt** [comment]
You can just insert the summary as a system message inside the chat history. Remember the message id as a variable, and delete it if you generate a new summary. That would be the easiest option I think.

If you want the summary to be "hidden", you can just remember the needed position in a variable, and the summary too, and then use `/inject` to make an ephemeral insert into the prompt, that you don't see. You can do that again and again, every time at the correct position, just by having an auto-executable QR that does this.

**Comment by Quairon-Nailo** [comment]
The problem with this solution is whatever variable I use to track the summary or the latest summarized message would remain the same between branches/timelines. Tracking different branches is something the summarizer already does by default, so if there was a macro I could use to get the ID of the last summarized message on the current branch, that would be enough, because I'd subtract that from the `{{lasstmessageID}}` and use `/inject` to automatically add the `{{summary}}` macro at the required depth.

**Comment by Wolfsblvt** [comment]
Currently only doable with workarounds, yeah. You can iterate over the messages in a loop, read the content, based on its role, then check if its empty, to see if it's a message from system. But that's a bit ugly.
Still, some users are doing named QR-closures and all that to easily query if a message is made by system, and such.

And if that workaround is too dirty, we should likely consider a side-request to this, and actually have a slash command that is able to search for message ids based on specific criteria.

**Comment by Quairon-Nailo** [comment]
I think I'm missing something, why are we looking for system messages? Is the summarize prompt saved as a system message? Sorry for the stupid question, I'm kinda new to STscript.

**Comment by Wolfsblvt** [comment]
Yes. Pretty sure the summary extension injects the summary into the prompt as a system message. Might be wrong though.
